sdk/lib/drivers/sound/mmixer/sup.c

```diff
@@ -685,47 +684,41 @@ MMixerSetGetVolumeControlDetails(
     if (!VolumeData)
         return MM_STATUS_UNSUCCESSFUL;
 
-    /* get input */
-    Input = (LPMIXERCONTROLDETAILS_UNSIGNED)MixerControlDetails->paDetails;
+    /* Get input */
+    Input = (LPMIXERCONTROLDETAILS_SIGNED)MixerControlDetails->paDetails;
     if (!Input)
-        return MM_STATUS_UNSUCCESSFUL; /* to prevent dereferencing NULL */
+        return MM_STATUS_UNSUCCESSFUL; /* To prevent dereferencing NULL */
 
-    if (bSet)
+    /* Loop for each channel */
+    for (Channel = 0; Channel < MixerControlDetails->cChannels; Channel++)
     {
-        /* FIXME SEH */
-        Value = Input->dwValue;
-        Index = Value / VolumeData->InputSteppingDelta;
-
-        if (Index >= VolumeData->ValuesCount)
+        if (bSet)
         {
-            DPRINT1("Index %u out of bounds %u \n", Index, VolumeData->ValuesCount);
-            return MM_STATUS_INVALID_PARAMETER;
-        }
+            /* FIXME SEH */
+            Index = Input[Channel].lValue / VolumeData->InputSteppingDelta;
 
-        Value = VolumeData->Values[Index];
-    }
+            if (Index >= VolumeData->ValuesCount)
+            {
+                DPRINT1("Index %u out of bounds %u \n", Index, VolumeData->ValuesCount);
+                return MM_STATUS_INVALID_PARAMETER;
+            }
 
-    /* set control details */
-    if (bSet)
-    {
-        /* TODO */
-        Status = MMixerSetGetControlDetails(MixerContext, MixerControl->hDevice, NodeId, bSet, KSPROPERTY_AUDIO_VOLUMELEVEL, 0, &Value);
-        Status = MMixerSetGetControlDetails(MixerContext, MixerControl->hDevice, NodeId, bSet, KSPROPERTY_AUDIO_VOLUMELEVEL, 1, &Value);
-    }
-    else
-    {
+            Value = VolumeData->Values[Index];
+        }
+
+        /* Get/set control details */
         Status = MMixerSetGetControlDetails(MixerContext, MixerControl->hDevice, NodeId, bSet, KSPROPERTY_AUDIO_VOLUMELEVEL, Channel, &Value);
-    }
 
-    if (!bSet)
-    {
-        dwValue = MMixerGetVolumeControlIndex(VolumeData, (LONG)Value);
-        /* FIXME SEH */
-        Input->dwValue = dwValue;
+        if (!bSet)
+        {
+            /* FIXME SEH */
+            Input[Channel].lValue = MMixerGetVolumeControlIndex(VolumeData, Value);
+        }
     }
-    else
+
+    if (bSet)
     {
-        /* notify clients of a line change  MM_MIXM_CONTROL_CHANGE with MixerControl->dwControlID */
+        /* Notify clients of a line change  MM_MIXM_CONTROL_CHANGE with MixerControl->dwControlID */
```

Comment:
```suggestion
        /* Notify clients of a line change MM_MIXM_CONTROL_CHANGE with MixerControl->dwControlID */
```
Not sure the `MM_MIXM_CONTROL_CHANGE with MixerControl->dwControlID` part is necessary, we can see this in the function call.